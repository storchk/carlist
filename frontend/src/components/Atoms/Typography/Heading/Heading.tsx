import { Typography } from '../Typography'
import type { HeadingProps } from './Heading.types'

const getHeadingByTag = (props: HeadingProps) => {
  switch (props.tag) {
    case 'h1':
      return (
        <Typography
          {...props}
          fontSize={{
            mobile: 'xl',
            tablet: 'xxl',
            desktop: 'xxxl',
          }}
        />
      )
    case 'h2':
      return (
        <Typography
          {...props}
          fontSize={{
            mobile: 'lg',
            tablet: 'xl',
          }}
          fontWeight="bold"
        />
      )
    case 'h3':
    default:
      return (
        <Typography
          {...props}
          fontSize={{
            mobile: 'lg',
          }}
          fontWeight="bold"
        />
      )
    case 'h4':
      return (
        <Typography
          {...props}
          fontSize={{
            mobile: 'lg',
          }}
          fontWeight="bold"
        />
      )
    case 'h5':
      return (
        <Typography
          {...props}
          fontSize={{
            mobile: 'md',
          }}
        />
      )
    case 'h6':
      return (
        <Typography
          {...props}
          fontSize={{
            mobile: 'xxs',
          }}
        />
      )
  }
}
export const Heading = ({ tag = 'h3', ...props }: HeadingProps): JSX.Element => {
  return getHeadingByTag({ tag, ...props })
}
