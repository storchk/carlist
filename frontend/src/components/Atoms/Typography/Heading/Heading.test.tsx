import { axe } from 'jest-axe'

import { testing } from '../../../../testing'
import { Heading } from './Heading'

const { render, renderer } = testing
const mockText = 'Lorem Ipsum'

describe('atoms: Typography/Heading', () => {
  it('snapshot: h1', () => {
    const view = renderer(<Heading tag="h1">{mockText}</Heading>)
    expect(view).toMatchSnapshot()
  })
  it('snapshot: h2', () => {
    const view = renderer(<Heading tag="h2">{mockText}</Heading>)
    expect(view).toMatchSnapshot()
  })
  it('snapshot: h3', () => {
    const view = renderer(<Heading tag="h3">{mockText}</Heading>)
    expect(view).toMatchSnapshot()
  })
  it('snapshot: h4', () => {
    const view = renderer(<Heading tag="h4">{mockText}</Heading>)
    expect(view).toMatchSnapshot()
  })
  it('snapshot: h5', () => {
    const view = renderer(<Heading tag="h5">{mockText}</Heading>)
    expect(view).toMatchSnapshot()
  })
  it('snapshot: h6', () => {
    const view = renderer(<Heading tag="h6">{mockText}</Heading>)
    expect(view).toMatchSnapshot()
  })
  it('a11y - No violations detected', async () => {
    const { container } = render(<Heading tag="h1">{mockText}</Heading>)
    await expect(axe(container)).resolves.toHaveNoViolations()
  })
})
