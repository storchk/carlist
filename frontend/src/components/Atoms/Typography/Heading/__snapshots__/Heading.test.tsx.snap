// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms: Typography/Heading snapshot: h1 1`] = `
.c0 {
  font-weight: 400;
  color: inherit;
  font-size: 24px;
}

@media (min-width: 768px) {
  .c0 {
    font-size: 28px;
  }
}

@media (min-width: 992px) {
  .c0 {
    font-size: 32px;
  }
}

<h1
  className="c0"
  fontSize={
    {
      "desktop": "xxxl",
      "mobile": "xl",
      "tablet": "xxl",
    }
  }
>
  Lorem Ipsum
</h1>
`;

exports[`atoms: Typography/Heading snapshot: h2 1`] = `
.c0 {
  font-weight: 700;
  color: inherit;
  font-size: 20px;
}

@media (min-width: 768px) {
  .c0 {
    font-size: 24px;
  }
}

<h2
  className="c0"
  fontSize={
    {
      "mobile": "lg",
      "tablet": "xl",
    }
  }
  fontWeight="bold"
>
  Lorem Ipsum
</h2>
`;

exports[`atoms: Typography/Heading snapshot: h3 1`] = `
.c0 {
  font-weight: 700;
  color: inherit;
  font-size: 20px;
}

<h3
  className="c0"
  fontSize={
    {
      "mobile": "lg",
    }
  }
  fontWeight="bold"
>
  Lorem Ipsum
</h3>
`;

exports[`atoms: Typography/Heading snapshot: h4 1`] = `
.c0 {
  font-weight: 700;
  color: inherit;
  font-size: 20px;
}

<h4
  className="c0"
  fontSize={
    {
      "mobile": "lg",
    }
  }
  fontWeight="bold"
>
  Lorem Ipsum
</h4>
`;

exports[`atoms: Typography/Heading snapshot: h5 1`] = `
.c0 {
  font-weight: 400;
  color: inherit;
  font-size: 18px;
}

<h5
  className="c0"
  fontSize={
    {
      "mobile": "md",
    }
  }
>
  Lorem Ipsum
</h5>
`;

exports[`atoms: Typography/Heading snapshot: h6 1`] = `
.c0 {
  font-weight: 400;
  color: inherit;
  font-size: 12px;
}

<h6
  className="c0"
  fontSize={
    {
      "mobile": "xxs",
    }
  }
>
  Lorem Ipsum
</h6>
`;
